skyportal/handlers/api/photometry_validation.py

```diff
@@ -0,0 +1,323 @@
+            required: true
+            schema:
+              type: int
+            description: Photometry ID
+        requestBody:
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  validated:
+                    type: boolean
+                    description: Whether the source is validated (True) or rejected (False)
+                required:
+                  - validated
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: int
+                              description: The id of the photomety_validation
+          400:
+            content:
+              application/json:
+                schema: Error
+
+        """
+        data = self.get_json()
+
+        validated = data.get('validated')
+        explanation = data.get('explanation')
+        notes = data.get('notes')
+
+        validator_instance = Validator()
+        params_to_be_validated = {
+            'method': 'POST',
+            'validated': validated,
+            'photometry_id': photometry_id,
+        }
+
+        if explanation is not None:
+            params_to_be_validated["explanation"] = explanation
+        if notes is not None:
+            params_to_be_validated["notes"] = notes
+
+        try:
+            validator = validator_instance.load(params_to_be_validated)
+        except ValidationError as e:
+            return self.error(f'Error parsing query params: {e.args[0]}.')
+
+        validated = validator['validated']
+        with self.Session() as session:
+            phot = session.scalars(
+                Photometry.select(session.user_or_token).where(
+                    Photometry.id == photometry_id
+                )
+            ).first()
+
+            if phot is None:
+                return self.error(
+                    f'Cannot find photometry point with ID: {photometry_id}.'
+                )
+
+            stmt = PhotometryValidation.select(session.user_or_token).where(
+                PhotometryValidation.photometry_id == photometry_id,
+            )
+            photometry_validation = session.scalars(stmt).first()
+            if photometry_validation:
+                # if the status and explanation are the same, do nothing
+                if (
+                    photometry_validation.validated == validated
+                    and photometry_validation.explanation == explanation
+                    and photometry_validation.notes == notes
+                ):
+                    return self.error(
```

Comment:
Necessary to error out in this case?

Reply:
What do you think @Theodlz ?

Reply:
Not necessary to throw an error no. Just send a success and don't modify anything in the DB!